internal/store/pod.go

```diff
@@ -274,6 +278,9 @@ var (
 					}
 				}
 
+				// compute the "Terminating" phase in order to reflect what a user can see with kubectl: https://github.com/kubernetes/kubernetes/blob/6a4216ba59ce1d09c8ac1c6229649bb71b7f6c85/pkg/printers/internalversion/printers.go#L737-L741
+				isPodTerminating := p.DeletionTimestamp != nil && p.Status.Reason != nodeUnreachablePodReason
```

Comment:
This is not really how kube state metrics is implemented otherwise. We usually just expose the raw data and leave computations like this to query time.

Reply:
I completely agree, but some logic was already added to compute the `Unknow status`. 
https://github.com/kubernetes/kube-state-metrics/pull/1013/files#diff-791ee1d484ea70de1ecd1d8875f8f5c0L286

Here we want to reflect what is displayed by "kubectl" to avoid confusion, and avoid having a Pod in a Running state when it is not.

Reply:
I think that was a mistake and since we're in the process of cleaning up kube-state-metrics and releasing preparing a breaking release, the time is right to change this. @LiliC @tariq1890 what do you think? We should definitely make sure that for these cases we have the required raw data available so we _can_ compute these things at runtime, but we should strive to remove any of these types of things from kube-state-metrics, just like we state [in our docs](https://github.com/kubernetes/kube-state-metrics#overview) :) .

Reply:
Yes agreed, we should expose just raw metrics, users can compute whatever they want afterwards, and/or create recording rules for this.

Reply:
Is there a way with the current information to know how many pods have the deletionTimestamp set?
With this info, we could indeed easily compute the "Terminating" status

Reply:
@brancz and @LiliC
I can rework this PR to add missing metrics for "raw" information to allow computing the `Terminating` and `Unknown` pod state from metrics.
Does that sound good?

Reply:
Yes that sounds perfect! :)

Reply:
PR updated with 2 new metrics:
- kube_pod_deleted
- kube_pod_status_reason

I can create another PR to remove the current logic for the `Unknown` pod phase.